src/Core/Domain/CartRule/Command/AddCartRuleCommand.php

```diff
@@ -213,82 +181,108 @@ public function getPriority(): int
     }
 
     /**
-     * @return bool
+     * @param int $priority
+     *
+     * @return AddCartRuleCommand
+     *
+     * @throws CartRuleConstraintException
      */
+    public function setPriority(int $priority): AddCartRuleCommand
+    {
+        if (0 >= $priority) {
+            throw new CartRuleConstraintException(
+                sprintf('Invalid cart rule priority "%s". Must be a positive integer.', $priority),
+                CartRuleConstraintException::INVALID_PRIORITY
+            );
+        }
+
+        $this->priority = $priority;
+
+        return $this;
+    }
+
     public function isActive(): bool
     {
-        return $this->isActive;
+        return $this->active;
     }
 
-    /**
-     * @return DateTime
-     */
-    public function getValidFrom(): DateTime
+    public function setActive(bool $active): AddCartRuleCommand
+    {
+        $this->active = $active;
+
+        return $this;
+    }
+
+    public function getValidFrom(): ?DateTime
     {
         return $this->validFrom;
     }
 
-    /**
-     * @return DateTime
-     */
-    public function getValidTo(): DateTime
+    public function getValidTo(): ?DateTime
     {
         return $this->validTo;
     }
 
-    /**
-     * @return int
-     */
+    public function setValidDateRange(DateTime $from, DateTime $to): AddCartRuleCommand
```

Comment:
```suggestion
    public function setDateRange(DateTime $from, DateTime $to): AddCartRuleCommand
```

Reply:
Actually now that you explained the purpose here I'd say:
```suggestion
    public function setValidityDateRange(DateTime $from, DateTime $to): AddCartRuleCommand
```